### DISCUSSION TITLE
Request: Add checkbox to hide custom track designs.

### DISCUSSION THREAD

**Discussion by MarcelVos96** [OpenRCT2/OpenRCT2#21177] [Feature requests]
I have dozens of custom track designs for some ride types, and finding the official ones can sometimes be a bit of a pain. A checkbox that hides custom ride designs would allow me and other people with the same issue to much easier see all the official ones.

This could function similarly to the option to show custom scenery objects and rides in the object selection screen, or the option to group by ride type in the ride build selection screen.

**Comment by pfroud** [comment]
I implemented a demo of this!!

<table>
<tr><td></td><td align="center">Show custom</td><td align="center">Hide custom</td></tr>
<tr>
<td>No filter</td>
<td><img src="https://github.com/OpenRCT2/OpenRCT2/assets/9056906/a81a019d-407b-4001-8807-52451bd22120" alt="show custom - no filter" width="300"></td>
<td><img src="https://github.com/OpenRCT2/OpenRCT2/assets/9056906/abfa08dc-265e-40d4-aee5-b4e989632ef7" alt="hide custom - no filter" width="300"></td>
</tr>
<tr>
<td>With filter</td>
<td><img src="https://github.com/OpenRCT2/OpenRCT2/assets/9056906/eb82f17f-face-4336-aae4-33a653251ccd" alt="show custom - with filter" width="300">
</td>
<td><img src="https://github.com/OpenRCT2/OpenRCT2/assets/9056906/f1ff3626-d07a-4021-bb20-eee17d5ef95a" alt="hide custom - with filter" width="300"></td>
</tr>
</table>

The checkbox is disabled in the Track Design Manager:
<img src="https://github.com/OpenRCT2/OpenRCT2/assets/9056906/32355461-8490-4b85-a50d-2a275195bb26" alt="checkbox disabled in track design manager" width="300">

I don't save track designs often so I don't know if this is good UI design. The checkbox could instead say "show custom ride designs" and be checked by default.

If no further discussion in a few days I will submit a pull request.

**Reply by tupaschoal** [reply]
I think I like the "show custom design" option with default true better. Should we just not show it in the truck designer?

Also is it being saved on the config and restored between different gaming sessions? I don't think this is a requirement, but I'm thinking whether it would be nice.

**Reply by ZehMatt** [reply]
Not sure this is ideal, I think we should do this a bit more like object selection to just have a few filters, I bet someone will come around asking for showing only custom designs at some point.

**Reply by spacek531** [reply]
Speedrunner rules allow for custom designs made only during the current session. If filters are added, I am sure that will be one of the requests.

**Reply by MarcelVos96** [reply]
Spacek asked me to clarify the rules for track designs in speedruns, so here it goes:

Any prebuilt that comes with the game is always allowed. If you play the OpenRCT2% category this includes RCT1 designs, even if it's an RCT2 scenario. You may build a custom design, save it, and then build that saves design in the same speedrun. If you do a speedrun of multiple levels in a row you may use any designs you saves during previous scenarios in the same run. Any custom design that you saved before the speedrun started is forbidden.

I always cleared my tracks folder (temporarily moved them all) before doing speedruns so that my already existing designs, which I can't use, aren't in the way. For speedruns that require only prebuilts that come with the game this feature will be quite useful as you wouldn't have to move your files, but for any run that requires custom designs even the reverse option (show only custom ones) wouldn't be that useful. We just name our designs "000" and they'll be at the top. It might save a second but not much more. It would still be a nice option to have though.

**Reply by pfroud** [reply]
How about a dropdown menu with three items:
- Show only custom track designs
- Show only built-in track designs
- Show all track designs

<img src="https://github.com/OpenRCT2/OpenRCT2/assets/9056906/e4026839-2b3e-4c47-8eb9-4928e45ffa9e" alt="dropdown menu to choose which track designs are visible" width="500">

**Reply by MarcelVos96** [reply]
I really like that, except I think that "show all track designs" should be at the top as it should be the default.

**Reply by karst** [reply]
It would be nice if we could also hide the contest track designs, so they're not clumped together with the somewhat decent default track designs. The filter menu looks like this in the object selector, which I think would be the right design for this, too.
![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/fdfa69e7-5037-42b8-bfa4-a0aaad1c0bcf)
It would mean more tags but I wouldn't mind this. Currently OpenRCT2 doesn't provide any official track designs >Yet< so it's not yet necessary.

I haven't done any research on what track design is provided by what, but we should at least have the following filters:
- Rollercoaster Tycoon
- Added Attractions / Loopy Landscapes
- Deluxe (Or a different name, these are all the u(####) files)
- RollerCoaster Tycoon 2
- Wacky Worlds
- Time Twister
- OpenRCT2 Official
- Custom

Ofcourse selected only/non-selected only would not make sense as a filter for this.

I will make some documentation on what track designs come with what: https://github.com/OpenRCT2/OpenRCT2/wiki/Track-designs

**Reply by MarcelVos96** [reply]
Tracks work differently, so all these options aren't necessary and only provide too much clutter.

For RCT2 a coaster type only got new designs in the expansions if it also got a new train design. So for tons of types (Side-friction, Virigina Reel, Looping coaster, wild mouse, mine train, etc) there simply aren't any TT/WW designs (and maybe no AA/LL designs either).

If we do want to do it like this then I think the three options of "RCT1",  "RCT2", and "Custom" should be enough. Sure, this doesn't allow you to only disable u-designs, but those are all grouped together and you can easily look over them anyway. It's already a fairly rare usecase as most coaster types don't have more than a couple u-designs to begin with. Many RCT1 designs have been repeated in RCT2 as well, making the need to look for just standard RCT1 designs even lower.

**Comment by ZehMatt** [comment]
> Tracks work differently, so all these options aren't necessary and only provide too much clutter.
> 
> For RCT2 a coaster type only got new designs in the expansions if it also got a new train design. So for tons of types (Side-friction, Virigina Reel, Looping coaster, wild mouse, mine train, etc) there simply aren't any TT/WW designs (and maybe no AA/LL designs either).
> 
> If we do want to do it like this then I think the three options of "RCT1", "RCT2", and "Custom" should be enough. Sure, this doesn't allow you to only disable u-designs, but those are all grouped together and you can easily look over them anyway. It's already a fairly rare usecase as most coaster types don't have more than a couple u-designs to begin with. Many RCT1 designs have been repeated in RCT2 as well, making the need to look for just standard RCT1 designs even lower.

I also think that is the best option, 3 filters, nice and simple. If it ever comes down to it we can potentially expand but I generally like the filter option more than the checkboxes.

**Comment by HtotheTML** [comment]
I also agree this option would be helpful, especially in the case of designs from RCT1 being reused in RCT2. Lot of duplicates there.

**Comment by autosysops** [comment]
I agree with this if the filter option is used as discussed later in this thread
